cloudquery: init at 6.11.2

mrgiles opened the pull request:
Add [CloudQuery](https://www.cloudquery.io/) to Nix Packages.

CloudQuery is a versatile [open-source](https://github.com/cloudquery/cloudquery) data movement tool built for developers that allows you to sync data from any [source](https://hub.cloudquery.io/plugins/source) to any [destination](https://hub.cloudquery.io/plugins/destination).
  
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>cloudquery</li>
  </ul>
</details>


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mrgiles committed:
cloudquery: init at 6.11.2

remove with lib

remove gen binary

mrgiles commented:
Hi @gepbird. Is this Ofborg queued test delay normal? I understand that there may be a shortage of Apple Silicon test machines but I'm not sure what is a reasonable time to wait, or if I should just run another commit/push to force a new Ofborg test run? Thanks!

```
OfBorg / cloudquery, cloudquery.passthru.tests on aarch64-darwin
Started 2d 53m 58s ago
```

gepbird commented:
> Hi @gepbird. Is this Ofborg queued test delay normal? I understand that there may be a shortage of Apple Silicon test machines but I'm not sure what is a reasonable time to wait, or if I should just run another commit/push to force a new Ofborg test run? Thanks!
> 
> ```
> OfBorg / cloudquery, cloudquery.passthru.tests on aarch64-darwin
> Started 2d 53m 58s ago
> ```

Yes, that is normal. If you really want to test it on darwin you can ping someone that has a darwin machine (I don't) and hope they respond. I'm not sure if pinging the entirety of darwin maintainers is a good idea but its an option.

Or you can skip testing it on darwin and hope it works, if people will be using this package on darwin and find some issues they will open a PR or an issue. Personally I don't push darwin testing for packages I maintain, but co-maintainers testing it on darwin is always a plus.

gepbird commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,51 @@
+    mv $out/bin/cli $out/bin/$pname
+  '';
+
+  meta = with lib; {
```
See https://github.com/NixOS/nixpkgs/issues/208242
```suggestion
  meta = {
```

mrgiles replied:
Fixed. Thanks for the link. I didn't understand why this was wrong, since it looks like the majority of the packages under `by-name` are currently using `with lib` in the `meta` section.

gepbird commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,51 @@
+  src = fetchFromGitHub {
+    owner = "cloudquery";
+    repo = "cloudquery";
+    rev = "cli-v${version}";
```
```suggestion
    rev = "refs/tags/cli-v${version}";
```

mrgiles replied:
Fixed. I think I saw this recommended somewhere before but somehow I didn't use it.

gepbird commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,51 @@
+  ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/$pname
```
```suggestion
    mv $out/bin/cli $out/bin/cloudquery
```

mrgiles replied:
Fixed. I copied that syntax from an existing package, Is it also an anti-pattern?

gepbird reviewed (changes requested):
Thanks for your contribution, code mostly LGTM, just a few nits below.

Also there's a  `gen` binary (which crashes but that's probably because I don't have any cloudquery data) and probably has a too generic name, consider renaming it to `cloudquery-gen`.

mrgiles commented:
> Also there's a gen binary (which crashes but that's probably because I don't have any cloudquery data) and probably has a too generic name, consider renaming it to cloudquery-gen.

I don't know how that `gen` binary got built. I compiled the source code by hand from source and it only produces the `cli` binary. I've also downloaded their pre-built binary and it doesn't include `gen`.

Somehow, the Nix builder creates it. If I can't find how it gets created, I could add a `postInstall` command to delete it: `rm $out/bin/gen`

Interestingly, searching for the issue upstream, I've found this: [feat(cli)!: Remove gen command](https://github.com/cloudquery/cloudquery/pull/2022). So I think this `gen` command currently servers no purpose and it can be safely deleted.

What do you think?

gepbird commented:
> > Also there's a gen binary (which crashes but that's probably because I don't have any cloudquery data) and probably has a too generic name, consider renaming it to cloudquery-gen.
> 
> I don't know how that `gen` binary got built. I compiled the source code by hand and it only produces the `cli` binary. I've also downloaded their pre-built binary and it doesn't include `gen`.
> 
> Somehow, the Nix builder creates it. If I can't find how it gets created, I could add a `postInstall` command to delete it: 
> 
> `rm $out/bin/gen`
> 
> Interestingly, searching for the issue upstream, I've found this: [feat(cli)!: Remove gen command](https://github.com/cloudquery/cloudquery/pull/2022). So I think this `gen` command currently serves no purpose and it can be safely deleted.
> 
> What do you think?

Yes, if upstream doesn't distribute it probably we shouldn't either. You can add that `postInstall` script.

mrgiles commented:
> Yes, if upstream doesn't distribute it probably we shouldn't either. You can add that postInstall script.

Removed `bin/gen`

> Please follow the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions), for a package init PR you can squash all your changes into a single commit.

Squashed commits.

gepbird reviewed (approved):
Thanks, LGTM!

mrgiles commented:
Thanks for reviewing!

mrgiles commented:
Hi @Aleksanaa, quick question: do we have to wait for the `passthru.tests on aarch64-darwin` to complete to be able to approve the PR? It's been queued for 7 days already. I'm just asking, no hurry. Thanks! 

```
OfBorg / cloudquery, cloudquery.passthru.tests on aarch64-darwin
Started 7d 8h 26m 16s ago
```

Aleksanaa commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  modRoot = "cli";
+
+  # Some checks fail to download deps
+  doCheck = false;
```
We can partially skip test, see https://nixos.org/manual/nixpkgs/unstable/#ssec-skip-go-tests

mrgiles replied:
Thanks for reviewing. I need to read over this and run some tests to see what works.

Aleksanaa replied:
Please add it back. It seems the test is a part of their workflow.

Aleksanaa commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,52 @@
+  postInstall = ''
+    mv $out/bin/cli $out/bin/cloudquery
+    rm $out/bin/gen
+  '';
```
```suggestion
  '' + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
    installShellCompletion --cmd cloudquery \
      --bash <($out/bin/cloudquery completion bash) \
      --fish <($out/bin/cloudquery completion fish) \
      --zsh <($out/bin/cloudquery completion zsh)
  '';
```

mrgiles replied:
I get build errors with the code above. 

Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

Aleksanaa replied:
> I get build errors with the code above.

Add `installShellFiles` to nativeBuildInputs.

> Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

https://github.com/cloudquery/cloudquery/blob/2f78b2b9d5c1b5438332443bbe6d4710c94f8a49/cli/.goreleaser.yaml#L96-L102

gepbird replied:
> I get build errors with the code above.

Probably you didn't put `stdenv` at the top or didn't put `installShellFiles` into `nativeBuildInputs`. Here's the documentation for installing shell completions: https://github.com/NixOS/nixpkgs/blob/d18ef8212246b78339007d745738aa4ce5d34230/doc/hooks/installShellFiles.section.md#installshellcompletion-installshellfiles-installshellcompletion

<details><summary>Full diff (click here to expand)</summary>

```diff
diff --git a/pkgs/by-name/cl/cloudquery/package.nix b/pkgs/by-name/cl/cloudquery/package.nix
index 3b245cf82414..d1ff8af97edf 100644
--- a/pkgs/by-name/cl/cloudquery/package.nix
+++ b/pkgs/by-name/cl/cloudquery/package.nix
@@ -2,6 +2,8 @@
   lib,
   buildGoModule,
   fetchFromGitHub,
+  installShellFiles,
+  stdenvNoCC,
 }:
 
 buildGoModule rec {
@@ -21,6 +23,10 @@ buildGoModule rec {
 
   modRoot = "cli";
 
+  nativeBuildInputs = [
+    installShellFiles
+  ];
+
   # Some checks fail to download deps
   doCheck = false;
 
@@ -30,10 +36,17 @@ buildGoModule rec {
     "-X github.com/cloudquery/cloudquery/cli/cmd.Version=${version}"
   ];
 
-  postInstall = ''
-    mv $out/bin/cli $out/bin/cloudquery
-    rm $out/bin/gen
-  '';
+  postInstall =
+    ''
+      mv $out/bin/cli $out/bin/cloudquery
+      rm $out/bin/gen
+    ''
+    + lib.optionalString (stdenvNoCC.buildPlatform.canExecute stdenvNoCC.hostPlatform) ''
+      installShellCompletion --cmd cloudquery \
+        --bash <($out/bin/cloudquery completion bash) \
+        --fish <($out/bin/cloudquery completion fish) \
+        --zsh <($out/bin/cloudquery completion zsh)
+    '';
 
   meta = {
     description = "Data movement tool to sync data from any source to any destination";
```
</details>

> Why do we need this? I built the original app from source and also downloaded the pre-built binary and neither include shell completion.

Of course the package works without shell completions, it's just an extra nice to have thing that is more convenient on NixOS. If you want to have shell completions on other distros, you'd have to put a line in your .bashrc/.zshrc or whatever shell you're using for every single program.

mrgiles replied:
The shell completions work fine. Thanks!

Aleksanaa commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,52 @@
+    "-w"
+    "-X github.com/cloudquery/cloudquery/cli/cmd.Version=${version}"
+  ];
+
```
```suggestion
  subPackages = [ "cmd" ];
```

Not sure

mrgiles replied:
If you use `subPackages = [ "cmd" ]`, `nixpkgs-review` throws this error: 
`go: go.mod file not found in current directory or any parent directory`

The directory with the `go.mod` file is under `cloudquery/cli/` (that's why I use the `modRoot = "cli"` statement).

gepbird replied:
I couldn't get this to work either. There exists a  `cli/go.mod` file, but no `cli/cmd/go.mod`. I assume in order to only build the `cli/cmd` go files, there needs to be a `cli/cmd/go.mod` but I'm not too familliar with golang.

Aleksanaa commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,52 @@
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/cloudquery
+    rm $out/bin/gen
```
```suggestion
```

Should not be needed if you set `subPackages`

mrgiles replied:
Still needed because `subPackages` doesn't work.

Aleksanaa replied:
Okay, ignore it

Aleksanaa commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,52 @@
+    mv $out/bin/cli $out/bin/cloudquery
+    rm $out/bin/gen
+  '';
+
```
```suggestion

  doInstallCheck = true;

  nativeInstallCheckInputs = [
    versionCheckHook
  ];
```

mrgiles replied:
With those lines above, `nixpkgs-review` threw the following error:

`error: undefined variable 'versionCheckHook'`

Aleksanaa replied:
> `error: undefined variable 'versionCheckHook'`

You have to add it to argument of this function of package definition. I don't need to make sure my diff can be run directly after applying. That's not how suggestions work.

gepbird replied:
> With those lines above, `nixpkgs-review` threw the following error:
> 
> `error: undefined variable 'versionCheckHook'`

You also need to import that and put it in `nativeBuildInputs`.

<details><summary>Full diff (click to expand)</summary>

```diff
diff --git a/pkgs/by-name/cl/cloudquery/package.nix b/pkgs/by-name/cl/cloudquery/package.nix
index 3b245cf82414..c7e0f150e849 100644
--- a/pkgs/by-name/cl/cloudquery/package.nix
+++ b/pkgs/by-name/cl/cloudquery/package.nix
@@ -2,6 +2,7 @@
   lib,
   buildGoModule,
   fetchFromGitHub,
+  versionCheckHook,
 }:
 
 buildGoModule rec {
@@ -21,6 +22,10 @@ buildGoModule rec {
 
   modRoot = "cli";
 
+  nativeBuildInputs = [
+    versionCheckHook
+  ];
+
   # Some checks fail to download deps
   doCheck = false;
 
@@ -35,6 +40,11 @@ buildGoModule rec {
     rm $out/bin/gen
   '';
 
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
   meta = {
     description = "Data movement tool to sync data from any source to any destination";
     mainProgram = "cloudquery";
```

See the docs for more information: https://github.com/NixOS/nixpkgs/blob/87c4f4d7ba9e21ff7decf22d6d3a11f8e5cf3b60/doc/hooks/versionCheckHook.section.md

mrgiles replied:
The checkPhase section still throws errors, here's the build log with the errors:
[cloudquery-x86_64-linux.log](https://github.com/user-attachments/files/18033612/cloudquery-x86_64-linux.log)

What do you recommend here? Thanks!

gepbird replied:
@mrgiles Those errors are coming from upstream's tests (`doCheck = false` to disable them).

Our version check (with `doInstallCheck = true` and `versionCheckHook` in `nativeInstallCheckInputs`) works.

<details><summary>Build log (with `doCheck = false`)</summary>

```
Using versionCheckHook
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/fiv59ma73fh8n09czqkzxlkhqk4a0iw0-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Building subPackage .
Building subPackage ./cmd
Building subPackage ./internal/analytics
Building subPackage ./internal/api
Building subPackage ./internal/auth
Building subPackage ./internal/docs
Building subPackage ./internal/enum
Building subPackage ./internal/env
Building subPackage ./internal/hub
Building subPackage ./internal/otel
Building subPackage ./internal/publish
Building subPackage ./internal/publish/images
Building subPackage ./internal/specs/v0
Building subPackage ./internal/specs/v0/gen
Building subPackage ./internal/tablenamechanger
Building subPackage ./internal/team
Building subPackage ./internal/transformer
Building subPackage ./internal/transformerpipeline
Building subPackage ./internal/uuid
buildPhase completed in 51 seconds
Running phase: installPhase
@nix { "action": "setPhase", "phase": "installPhase" }
Running phase: fixupPhase
@nix { "action": "setPhase", "phase": "fixupPhase" }
shrinking RPATHs of ELF executables and libraries in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2
shrinking /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
checking for references to /build/ in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2...
patchelf: cannot find section '.dynamic'. The input file is most likely statically linked
patching script interpreter paths in /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2
stripping (with command strip and flags -S -p) in  /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin
Running phase: installCheckPhase
@nix { "action": "setPhase", "phase": "installCheckPhase" }
Executing versionCheckPhase
Did not find version 6.11.2 in the output of the command /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery --help
CloudQuery CLI

High performance data integration at scale.

Find more information at:
	https://www.cloudquery.io

Usage:
  cloudquery [command]

Available Commands:
  addon           Addon commands
  init            Generate a configuration file for a sync
  login           Login to CloudQuery Hub.
  logout          Log out of CloudQuery Hub.
  migrate         Update schema of your destinations based on the latest changes in sources from your configuration
  plugin          Plugin commands
  switch          Switches between teams.
  sync            Sync resources from configured source plugins to destinations
  tables          Generate documentation for all supported tables of source plugins specified in the spec(s)
  test-connection Test plugin connections to sources and/or destinations
  validate-config Validate config

Flags:
      --cq-dir string            directory to store cloudquery files, such as downloaded plugins (default ".cq")
  -h, --help                     help for cloudquery
      --invocation-id uuid       useful for when using Open Telemetry integration for tracing and logging to be able to correlate logs and traces through many services (default <NEW-RANDOM-UUID>)
      --log-console              enable console logging
      --log-file-name string     Log filename (default "cloudquery.log")
      --log-format string        Logging format (json, text) (default "text")
      --log-level string         Logging level (trace, debug, info, warn, error) (default "info")
      --no-log-file              Disable logging to file
      --telemetry-level string   Telemetry level (none, errors, stats, all) (default "all")
  -v, --version                  version for cloudquery

Use "cloudquery [command] --help" for more information about a command.
Successfully managed to find version 6.11.2 in the output of the command /nix/store/4vyjdwpcqkf7bi6v22vfyzgdrymm3f74-cloudquery-6.11.2/bin/cloudquery --version
cloudquery version 6.11.2
Finished versionCheckPhase
no Makefile or custom installCheckPhase, doing nothing
```
</details>

mrgiles commented:
Thank you so much for your time in reviewing and sharing your knowledge with me @Aleksanaa and @gepbird.
I will go over your comments and test each of the suggestions.
It's getting late here now but I will continue working on it tomorrow.
Thanks!

mrgiles committed:
add shell completion

mrgiles committed:
add installCheck

gepbird commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,69 @@
+        --zsh <($out/bin/cloudquery completion zsh)
+    '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
```
After updating to 6.12.9 `--version` will print `cloudquery version development`, making the build fail with `versionCheckHook` because it can't find the the version number. I guess by adding some `ldflags` this can be fixed.

gepbird commented on pkgs/by-name/cl/cloudquery/package.nix:
```diff
@@ -0,0 +1,69 @@
+  src = fetchFromGitHub {
+    owner = "cloudquery";
+    repo = "cloudquery";
+    rev = "refs/tags/cli-v${version}";
```
Nit. For this you need to rebase onto master.
```suggestion
    tag = "cli-v${version}";
```

gepbird reviewed (changes requested):
Hi @mrgiles, just a friendly ping after 2 weeks to fix the package not building :)

I've left some other improvement that are not necessary, just nice to have.

And again, please squash your commits.
